src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.MovieSessionDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.MovieSession;
+import com.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query = session.createQuery("from MovieSession where movie.id "
+                    + "= :movieId and showTime >= :dateFrom and showTime < :dateTo",
+                    MovieSession.class);
+            query.setParameter("movieId", movieId);
+            LocalDateTime dateFrom = date.atStartOfDay();
+            query.setParameter("dateFrom", dateFrom);
+            LocalDateTime dateTo = dateFrom.plusHours(24L);
+            query.setParameter("dateTo", dateTo);
+            return query.getResultList();
+        }
```

Comment:
same here, it would be better to catch exception

Reply:
fixed